git/git.go

```diff
@@ -56,9 +56,16 @@ func ShowRefs(ref ...string) ([]Ref, error) {
 
 // CurrentBranch reads the checked-out branch for the git repository
 func CurrentBranch() (string, error) {
-	refCmd := GitCommand("symbolic-ref", "--quiet", "--short", "HEAD")
+	// Available from Git 2.22 and above
+	branchCmd := GitCommand("branch", "--show-current")
+	output, err := run.PrepareCmd(branchCmd).Output()
+
+	// Fallback to symbolic-ref
+	if err != nil {
+		refCmd := GitCommand("symbolic-ref", "--quiet", "--short", "HEAD")
```

Comment:
instead of using --short, since it has correctness issues, we can just do what `--short` does internally and strip off the prefix `refs/head/`

Reply:
Ah, I didn't understand this point from @nightpool before, but now I do. Thank you!

@mmontes11 could you revert back to the previous approach with `git symbolic-ref`, but just omit `--short` since that flag is what is causing the bug? Then after the command runs, we can strip off `refs/heads/` before. I think that would definitely be a simpler approach, and simpler is always better. Sorry to have sent you down a wrong path! 🙇

Reply:
@nightpool now I get your point as well

@mislav seems a better approach, rather than relying on fallbacks

Reply:
Ah, sorry about being unclear earlier! Glad to have helped :D